cpp/src/io/statistics/statistics.cuh

```diff
@@ -83,12 +86,37 @@ struct string_stats {
   }
 };
 
+struct byte_array_stats {
```

Comment:
I suspect you considered this already: Since we're leaning into the similarity to `string_stats`, should we combine the two? 
```c++
template <typename T> struct t_array_stats {...};
struct string_stats : t_array_stats<char> {};
struct byte_array_stats : t_array_stats<int8_t> {};
```
Using `typedef` might obfuscate debugging, but the above might *possibly* work.

Reply:
This would further ensure they remain in step. This isn't a bad suggestion at all.